core/src/main/java/org/mule/routing/correlation/EventCorrelator.java

```diff
@@ -415,8 +418,17 @@ protected void handleGroupExpiry(EventGroup group) throws MessagingException
                 logger.warn("Failed to clear group with id " + group.getGroupId()
                             + " since underlying ObjectStore threw Exception:" + e.getMessage());
             }
-            throw new CorrelationTimeoutException(CoreMessages.correlationTimedOut(group.getGroupId()),
-                                                  groupCollectionEvent);
+            
+            // This is intended to avoid an error in the handling of the correlation exception
+            // in case no event group was created before the timeout (no message for aggregate
+            // were received yet). A "custom" mule event is created so that a message is available.
+            if (groupCollectionEvent instanceof VoidMuleEvent)
+            {
+                groupCollectionEvent =
+                        new DefaultMuleEvent(new DefaultMuleMessage("No Group Collection Event was created before the timeout", muleContext), MessageExchangePattern.ONE_WAY, flowConstruct);
```

Comment:
use a NullPayload

Reply:
No static import to make it clear in NullPayload.getInstance().